src/dune_rules/melange.mli

```diff
@@ -34,3 +34,5 @@ module Cm_kind : sig
     val make_all : 'a -> 'a t
   end
 end
+
+val make_module_name : Module.t -> string
```

Comment:
First, I think we should move this function to module_compilation. It's the only place where it's used so that's where it belongs.

I would also rename it into something a bit more descriptive. For example, `js_basename`.

The return type of this function can also be `Filename.t` which will clarify the intent a little.

Reply:
It is used on both `Melange_rules` and `Module_compilation`.

Changed name and returned type in https://github.com/ocaml/dune/pull/6400/commits/7b982c0409e4d4240a2da32498fee7cbd03cf139.